programs/bench.h

```diff
@@ -19,46 +19,140 @@ extern "C" {
 #define ZSTD_STATIC_LINKING_ONLY   /* ZSTD_compressionParameters */
 #include "zstd.h"     /* ZSTD_compressionParameters */
 
+typedef enum {
+    BMK_timeMode = 0,
+    BMK_iterMode = 1
+} BMK_loopMode_t;
+
+typedef enum {
+    BMK_both = 0,
+    BMK_decodeOnly = 1,
+    BMK_compressOnly = 2
+} BMK_mode_t;
+
+#define ERROR_STRUCT(baseType, typeName) typedef struct { \
```

Comment:
What should I do instead of this? Since exiting the program is undesirable now, passing out an error result seems necessary. Should error just be manually written as a field of `BMK_result_t` and `BMK_customResult_t`?

Reply:
This construction is correct.

Just, writing it as a macro like this, implies that you are going to use this scheme in multiple places. Which is effectively the case in following code.

I initially thought you would need only to have one case like this, 
and maybe I'm wrong on this point.

Bottom line : if you need it only once, it's more straightforward to write the same thing but directly.

The difference is for the reader of the API : it's easier to understand the relation between the 2 types for a newcomer when they are written directly.
Here, doing the same thing but through a magic macro, it takes time to understand what is going on.
At the very least, I believe it would deserve some more detailed instructions in code comment.